### PR TITLE
Improve interoperability with metadata XML using default namespaces

### PR DESCRIPTION
**Pull request by weiiwang01**
The `same-integrator-operator` currently experiences issues when processing [metadata XML that uses default namespaces](https://github.com/canonical/saml-integrator-operator/blob/fix-xml-ns/tests/unit/files/metadata_default_namespaces.xml). This pull request addresses this issue by providing predefined namespace prefix mapping values, aligning with the [SAML 2.0 standard](https://docs.oasis-open.org/security/saml/v2.0/saml-profiles-2.0-os.pdf).

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]
LGTM. Thanks!

File: tests/unit/files/metadata_default_namespaces.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes128-cbc"></EncryptionMethod>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes192-cbc"></EncryptionMethod>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes256-cbc"></EncryptionMethod>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#rsa-oaep-mgf1p"></EncryptionMethod>
+    </KeyDescriptor>
+    <NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:transient</NameIDFormat>
+    <SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://saml.canonical.test/sso"></SingleSignOnService>
+    <SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://saml.canonical.test/sso"></SingleSignOnService>
+  </IDPSSODescriptor>
+</EntityDescriptor>
```
Comment (yanksyoon) [inline tests/unit/files/metadata_default_namespaces.xml:25]: I think this file is missing a final new line

**Review by yanksyoon — APPROVED** [review]
LGTM, just a minor missing final new line on the xml file! :D

File: tests/unit/files/metadata_default_namespaces.xml (hunk 1 above)
Comment (weiiwang01) [reply tests/unit/files/metadata_default_namespaces.xml:25]: Yes, but this file is returned from the IdP server, so I just want to keep it as is.

**Pull request merged**